### PR TITLE
TEST: Fix data_finiliaze in test_bcast

### PR DESCRIPTION
**Pull request by lappazos**
## What
Fix free in test_bcast 

## Why ?
Un-needed free

## How ?
Add same condition as in allocation

### CODE REVIEW

**Comment by lappazos** [comment]
bot:retest

**Comment by vspetrov** [comment]
@lappazos why do we need this PR? i saw this change in @shimmybalsam PR on mpool.

**Comment by lappazos** [comment]
> @lappazos why do we need this PR? i saw this change in @shimmybalsam PR on mpool.

@vspetrov just because it's a separate bug, to keep order. @shimmybalsam will simply rebase and solve the conflict

File: test/gtest/core/test_bcast.cc [hunk 1]
```diff
@@ -54,10 +54,13 @@ class test_bcast : public UccCollArgs, public ucc::test
...
     void data_fini(UccCollCtxVec ctxs)
     {
-        for (gtest_ucc_coll_ctx_t* ctx : ctxs) {
+        for (auto r = 0; r < ctxs.size(); r++) {
+            gtest_ucc_coll_ctx_t *ctx = ctxs[r];
             ucc_coll_args_t* coll = ctx->args;
             UCC_CHECK(ucc_mc_free(coll->src.info.buffer, mem_type));
-            UCC_CHECK(ucc_mc_free(ctx->init_buf, UCC_MEMORY_TYPE_HOST));
+            if (r == coll->root) {
+                UCC_CHECK(ucc_mc_free(ctx->init_buf, UCC_MEMORY_TYPE_HOST));
```
Comment (vspetrov) [inline test/gtest/core/test_bcast.cc:62]: If we are to fix, then lets do it right: the init_buf array is allocated always on host (UCC_MEMORY_TYPE_HOST is provided). We should not use ucc_mc_alloc in this case, but rather use ucc_malloc (and hence ucc_free).

**Review by vspetrov — CHANGES_REQUESTED** [review]

File: test/gtest/core/test_bcast.cc (hunk 1 above)
Comment (lappazos) [reply test/gtest/core/test_bcast.cc:62]: in another PR

**Review by vspetrov — APPROVED** [review]

**Review by Sergei-Lebedev — APPROVED** [review]

**Pull request merged**